src/com/lilithsthrone/controller/CoveringController.java

```diff
@@ -0,0 +1,461 @@
+package com.lilithsthrone.controller;
+
+import org.w3c.dom.events.EventTarget;
+
+import com.lilithsthrone.game.character.body.coverings.AbstractBodyCoveringType;
+import com.lilithsthrone.game.character.body.coverings.BodyCoveringType;
+import com.lilithsthrone.game.character.body.coverings.Covering;
+import com.lilithsthrone.game.character.body.valueEnums.BodyHair;
+import com.lilithsthrone.game.character.body.valueEnums.CoveringModifier;
+import com.lilithsthrone.game.character.body.valueEnums.CoveringPattern;
+import com.lilithsthrone.game.character.body.valueEnums.HairLength;
+import com.lilithsthrone.game.character.body.valueEnums.HairStyle;
+import com.lilithsthrone.game.character.body.valueEnums.PiercingType;
+import com.lilithsthrone.game.dialogue.DialogueNodeType;
+import com.lilithsthrone.game.dialogue.places.dominion.lilayashome.RoomPlayer;
+import com.lilithsthrone.game.dialogue.places.dominion.shoppingArcade.SuccubisSecrets;
+import com.lilithsthrone.game.dialogue.places.dominion.slaverAlley.ScarlettsShop;
+import com.lilithsthrone.game.dialogue.responses.Response;
+import com.lilithsthrone.game.dialogue.utils.BodyChanging;
+import com.lilithsthrone.game.dialogue.utils.CharacterModificationUtils;
+import com.lilithsthrone.game.dialogue.utils.MiscDialogue;
+import com.lilithsthrone.main.Main;
+import com.lilithsthrone.utils.colours.Colour;
+import com.lilithsthrone.utils.colours.PresetColour;
+
+public class CoveringController {
+	private static final boolean noCost = !Main.game.isInNewWorld()
+			|| ScarlettsShop.isSlaveCustomisationMenu()
+			|| Main.game.getCurrentDialogueNode() == MiscDialogue.getMakeupDialogueForEqualityCheck()
+			|| Main.game.getCurrentDialogueNode() == RoomPlayer.AUNT_HOME_PLAYERS_ROOM_MAKEUP
+			|| Main.game.getCurrentDialogueNode().getDialogueNodeType() == DialogueNodeType.PHONE;
```

Comment:
This looks wrong; was this variable supposed to have been inside (or a function called from) each of the `init*Listeners`?

Reply:
`noCost` has not been changed as part of this rework
[See here](https://github.com/Innoxia/liliths-throne-public/blob/dev/src/com/lilithsthrone/controller/MainControllerInitMethod.java#L4936)
It is checked about 2 dozen times in various listeners, could be made into a method call

Reply:
Changed to method call